[#55523] Add Dialog not mobile friendly: don't center align on mobile

judithroth opened the pull request:
Before (mobile / desktop):
![Screenshot from 2024-06-13 17-18-57](https://github.com/opf/openproject/assets/9654673/b18e3287-9dd6-4f8f-bc30-88640b20564d)
![Screenshot from 2024-06-13 17-19-18](https://github.com/opf/openproject/assets/9654673/ec9a2a62-41f2-4bae-9235-f69e66763629)

After (mobile / desktop):
![Screenshot from 2024-06-13 17-18-33](https://github.com/opf/openproject/assets/9654673/f194691c-142f-454c-8991-dea1fbd56b9b)
![Screenshot from 2024-06-13 17-21-36](https://github.com/opf/openproject/assets/9654673/8b121f43-4035-480c-86f5-6573abdbd02e)

judithroth committed:
[#55523] Add Dialog not mobile friendly: don't center align on mobile

HDinger reviewed (changes requested):
That class name is kind of complicated. Nobody will understand why there is this limitation because normally you want an element in all screens center aligned. The only reason we do not want that here, is because we change the flex-direction with the other class. In that case, this class should then also be responsible for the correct alignment.
So I'd suggest something like this:

```sass
// For some specific cases like progress form in small screens,
// we need to change the direction of elements to column,
// therefore elements are placed below each other
.FormControl-horizontalGroup--sm-vertical
  @media screen and (max-width: $breakpoint-sm)
    .FormControl-horizontalGroup
      flex-direction: column
      row-gap: 1rem
      // Enforce the alignment due to lower specificity
      align-items: normal !important

.FormControl-horizontalGroup--center-aligned
  .FormControl-horizontalGroup
    align-items: center
```

judithroth committed:
[#55523] Add Dialog not mobile friendly: don't center align on mobile (Review feedback)

This is yet another solution for the problem which Henriette and I
agreed on. No complex naming and no important! involved in this
variant.
